tests/linear_tree/test_lt_formulation.py

```diff
@@ -0,0 +1,728 @@
+    for i in splits[0].keys():
+        for key in splits[0][i].keys():
+            assert key in splits_key_list
+    # test for leaves
+    # assert the number of leaves
+    assert len(leaves[0].keys()) == 6
+    # assert whether all the dicts have such keys
+    leaves_key_list = [
+        "loss",
+        "samples",
+        "models",
+        "slope",
+        "intercept",
+        "parent",
+        "bounds",
+    ]
+    for j in leaves[0].keys():
+        for key in leaves[0][j].keys():
+            assert key in leaves_key_list
+            # if the key is slope, test the shape of it
+            if key == "slope":
+                assert len(leaves[0][j][key]) == n_inputs
+            # elif the key is bounds, test the lb <= ub
+            elif key == "bounds":
+                features = leaves[0][j][key].keys()
+                for k in range(len(features)):
+                    lb = leaves[0][j][key][k][0]
+                    ub = leaves[0][j][key][k][1]
+                    # there is chance that don't have lb and ub at this step
+                    if lb is not None and ub is not None:
+                        assert lb <= ub
+    # test for thresholds
+    # assert whether each feature has threshold
+    assert len(thresholds[0].keys()) == n_inputs
+    # assert the number of thresholds
+    thresholds_count = 0
+    for k in range(len(thresholds[0].keys())):
+        for _ in range(len(thresholds[0][k].keys())):
+            thresholds_count += 1
+    assert thresholds_count == len(splits[0].keys())
+
+
+@pytest.mark.skipif(
+    not lineartree_available or not cbc_available,
+    reason="Need Linear-Tree Package and cbc",
+)
+def test_bigm_formulation_multi_var():
+    regr = linear_model_tree(X=X, y=Y)
+    input_bounds = {0: (min(X[:, 0]), max(X[:, 0])), 1: (min(X[:, 1]), max(X[:, 1]))}
+    ltmodel_small = LinearTreeDefinition(regr, unscaled_input_bounds=input_bounds)
+    formulation1_lt = LinearTreeGDPFormulation(ltmodel_small, transformation="bigm")
+
+    model1 = pe.ConcreteModel()
+    model1.x0 = pe.Var(initialize=0)
+    model1.x1 = pe.Var(initialize=0)
+    model1.y = pe.Var(initialize=0)
+    model1.obj = pe.Objective(expr=1)
+    model1.lt = OmltBlock()
+    model1.lt.build_formulation(formulation1_lt)
+
+    @model1.Constraint()
+    def connect_input1(mdl):
+        return mdl.x0 == mdl.lt.inputs[0]
+
+    @model1.Constraint()
+    def connect_input2(mdl):
+        return mdl.x1 == mdl.lt.inputs[1]
+
+    @model1.Constraint()
+    def connect_outputs(mdl):
+        return mdl.y == mdl.lt.outputs[0]
+
+    model1.x0.fix(0.5)
+    model1.x1.fix(0.8)
+
+    status_1_bigm = pe.SolverFactory("cbc").solve(model1, tee=True)
+    pe.assert_optimal_termination(status_1_bigm)
+    solution_1_bigm = pe.value(model1.y)
+    y_pred = regr.predict(
+        np.array([pe.value(model1.x0), pe.value(model1.x1)]).reshape(1, -1)
+    )
+    assert y_pred[0] - solution_1_bigm <= 1e-4
```

Comment:
Like above. Use Pytest.approx. (Check for others).

Reply:
Changed all to use pytest.approx!